diesel/src/query_dsl/filter_dsl.rs

```diff
@@ -88,19 +88,20 @@ use helper_types::FindBy;
 /// assert_eq!(Err::<(i32, String), _>(NotFound), users.find(3).first(&connection));
 /// # }
 /// ```
-pub trait FindDsl<PK> where
-    Self: Table + FilterDsl<Eq<<Self as Table>::PrimaryKey, PK>>,
-    PK: AsExpression<<Self::PrimaryKey as Expression>::SqlType>,
-    Eq<Self::PrimaryKey, PK>: SelectableExpression<Self, SqlType=Bool> + NonAggregate,
-{
-    fn find(self, id: PK) -> FindBy<Self, Self::PrimaryKey, PK> {
-        let primary_key = self.primary_key();
-        self.filter(primary_key.eq(id))
-    }
+pub trait FindDsl<PK> {
+    type Output: AsQuery;
+
+    fn find(self, id: PK) -> Self::Output;
 }
 
 impl<T, PK> FindDsl<PK> for T where
     T: Table + FilterDsl<Eq<<T as Table>::PrimaryKey, PK>>,
     PK: AsExpression<<T::PrimaryKey as Expression>::SqlType>,
-    Eq<T::PrimaryKey, PK>: SelectableExpression<T, SqlType=Bool> + NonAggregate,
-{}
+{
+    type Output = FindBy<Self, T::PrimaryKey, PK>;
```

Comment:
`FindBy` is just a helper for `Filter<Source, Eq<Column, Value>>` which was common enough to justify a slightly more concise alias.